model_analyzer/result/model_constraints.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2021-2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class ModelConstraints:
+    """
+    A class representing the Constraints used for a single model.
+    """
+
+    def __init__(self, constraints):
+        """
+        Parameters
+        ----------
+        constraints: dict
+            keys are metrics and values are 
+            constraint_type: constraint_value pairs
```

Comment:
Please help us with more details, unable to understand the requirement comment.

Reply:
```
constraints: dict
            keys are metrics and values are 
            constraint_type: constraint_value pairs
```
The key is the name of a metric (a string), not an actual metric.


The 'values' are `constraint_type:constraint_value pairs`. In reality `constraint_type` here is a string. However, I believe the only legal possibilities for that string are `min` and `max`. I'm wondering if we should use an enumerator (or python equivalent) instead of a string if there are only two legal values for `constraint_type`.

Reply:
As discussed, schema validation is happening before creating this object, and will continue with the current approach. Modified the description as suggested.